pallets/confidential-docs/src/lib.rs

```diff
@@ -54,95 +53,102 @@ pub mod pallet {
 	pub struct Pallet<T>(_);
 
-
 	#[pallet::storage]
 	#[pallet::getter(fn vaults)]
 	pub(super) type Vaults<T: Config> = StorageMap<
 		_,
 		Blake2_256,
 		UserId, //user identifier
 		Vault<T>,
-		OptionQuery
+		OptionQuery,
 	>;
 
-  #[pallet::storage]
+	#[pallet::storage]
 	#[pallet::getter(fn public_keys)]
-	pub(super) type PublicKeys<T: Config> = StorageMap<
-		_,
-		Blake2_256,
-		T::AccountId,
-		PublicKey,
-		OptionQuery
-	>;
+	pub(super) type PublicKeys<T: Config> =
+		StorageMap<_, Blake2_256, T::AccountId, PublicKey, OptionQuery>;
 
 	#[pallet::storage]
 	#[pallet::getter(fn users_ids)]
-	pub(super) type UserIds<T: Config> = StorageMap<
-		_,
-		Identity,
-		[u8; 32],
-		UserId,
-		OptionQuery,
-	>;
+	pub(super) type UserIds<T: Config> = StorageMap<_, Identity, [u8; 32], UserId, OptionQuery>;
 
 	#[pallet::storage]
 	#[pallet::getter(fn owned_docs)]
-	pub(super) type OwnedDocs<T: Config> = StorageMap<
-		_,
-		Blake2_256,
-		CID,
-		OwnedDoc<T>,
-		OptionQuery
-	>;
+	pub(super) type OwnedDocs<T: Config> = StorageMap<_, Blake2_256, CID, OwnedDoc<T>, OptionQuery>;
 
 	#[pallet::storage]
 	#[pallet::getter(fn owned_docs_by_owner)]
-	pub(super) type OwnedDocsByOwner<T: Config> = StorageMap<
-		_,
-		Blake2_256,
-		T::AccountId,
-		BoundedVec<CID, T::MaxOwnedDocs>,
-		ValueQuery
-	>;
+	pub(super) type OwnedDocsByOwner<T: Config> =
+		StorageMap<_, Blake2_256, T::AccountId, BoundedVec<CID, T::MaxOwnedDocs>, ValueQuery>;
 
 	#[pallet::storage]
 	#[pallet::getter(fn shared_docs)]
-	pub(super) type SharedDocs<T: Config> = StorageMap<
+	pub(super) type SharedDocs<T: Config> =
+		StorageMap<_, Blake2_256, CID, SharedDoc<T>, OptionQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn shared_docs_by_to)]
+	pub(super) type SharedDocsByTo<T: Config> =
+		StorageMap<_, Blake2_256, T::AccountId, BoundedVec<CID, T::MaxSharedToDocs>, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn shared_docs_by_from)]
+	pub(super) type SharedDocsByFrom<T: Config> =
+		StorageMap<_, Blake2_256, T::AccountId, BoundedVec<CID, T::MaxSharedFromDocs>, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn groups)]
+	pub(super) type Groups<T: Config> = StorageMap<
 		_,
 		Blake2_256,
-		CID,
-		SharedDoc<T>,
+		// Group Account Id
+		T::AccountId,
+		Group<T>,
 		OptionQuery,
 	>;
 
 	#[pallet::storage]
-	#[pallet::getter(fn shared_docs_by_to)]
-	pub(super) type SharedDocsByTo<T: Config> = StorageMap<
+	#[pallet::getter(fn group_members)]
+	pub(super) type GroupMembers<T: Config> = StorageDoubleMap<
 		_,
 		Blake2_256,
+		// Group Account Id
 		T::AccountId,
-		BoundedVec<CID, T::MaxSharedToDocs>,
-		ValueQuery
+		Blake2_256,
+		// Member Account Id
+		T::AccountId,
+		GroupMember<T>,
+		OptionQuery,
 	>;
 
 	#[pallet::storage]
-	#[pallet::getter(fn shared_docs_by_from)]
-	pub(super) type SharedDocsByFrom<T: Config> = StorageMap<
+	#[pallet::getter(fn member_groups)]
+	pub(super) type MemberGroups<T: Config> = StorageMap<
```

Comment:
- Is this storage map used to know the groups to which a user belongs?

- How can I find out which members belong to a group? Given a group id, I want to know the list of group members.